libs/langchain/langchain/schema/runnable/history.py

```diff
@@ -0,0 +1,254 @@
+        if super_schema.__custom_root_type__ is not None:
+            from langchain.schema.messages import BaseMessage
+
+            fields: Dict = {}
+            if self.input_messages_key:
+                fields[self.input_messages_key] = (str, ...)
+            else:
+                fields["__root__"] = (Sequence[BaseMessage], ...)
+            if self.message_history_key:
+                fields[self.message_history_key] = (Sequence[BaseMessage], ...)
+            return create_model(  # type: ignore[call-overload]
+                "RunnableWithChatHistoryInput",
+                **fields,
+            )
+        else:
+            return super_schema
+
+    def _get_input_messages(
+        self, input_val: Union[str, BaseMessage, Sequence[BaseMessage]]
+    ) -> List[BaseMessage]:
+        from langchain.schema.messages import BaseMessage
+
+        if isinstance(input_val, str):
+            from langchain.schema.messages import HumanMessage
+
+            return [HumanMessage(content=input_val)]
+        elif isinstance(input_val, BaseMessage):
+            return [input_val]
+        elif isinstance(input_val, (list, tuple)):
+            return list(input_val)
+        else:
+            raise ValueError(
+                f"Expected str, BaseMessage, List[BaseMessage], or Tuple[BaseMessage]. "
+                f"Got {input_val}."
+            )
+
+    def _get_output_messages(
+        self, output_val: Union[str, BaseMessage, Sequence[BaseMessage], dict]
+    ) -> List[BaseMessage]:
+        from langchain.schema.messages import BaseMessage
+
+        if isinstance(output_val, dict):
+            output_val = output_val[self.output_messages_key or "output"]
+
+        if isinstance(output_val, str):
+            from langchain.schema.messages import AIMessage
+
+            return [AIMessage(content=output_val)]
+        elif isinstance(output_val, BaseMessage):
+            return [output_val]
+        elif isinstance(output_val, (list, tuple)):
+            return list(output_val)
+        else:
+            raise ValueError()
+
+    def _enter_history(self, input: Any, config: RunnableConfig) -> List[BaseMessage]:
+        hist: BaseChatMessageHistory = config["configurable"]["message_history"]
+        # return only historic messages
+        if self.message_history_key:
+            return hist.messages.copy()
+        # return all messages
+        else:
+            input_val = (
+                input if not self.input_messages_key else input[self.input_messages_key]
+            )
+            return hist.messages.copy() + self._get_input_messages(input_val)
+
+    async def _aenter_history(
+        self, input: Dict[str, Any], config: RunnableConfig
+    ) -> List[BaseMessage]:
+        return await asyncio.get_running_loop().run_in_executor(
+            None, self._enter_history, input, config
+        )
+
+    def _exit_history(self, run: Run, config: RunnableConfig) -> None:
```

Comment:
Because this is handled by a tracer any exception in this code will be silently ignored. I don't think this is the desired behavior by default?